### DISCUSSION TITLE
Using a Crazyflie with a Raspberry Pi Zero and the Thrust Upgrade Bundle?

### DISCUSSION THREAD

**Discussion by shreeram-murali** [bitcraze/discussions#552] [Q&A] [answered]
Hi!

I’ve got my hands on the new thrust upgrade bundle and I am planning on adding a Raspberry Pi Zero (and a Raspicam) to the Crazyflie. I have a few questions, regarding the weight, and powering it up. 

1. The Pi Zero requires 5V, but I’ve read that it can work at as low as 3.6V. But the single Li-Po cell would output anywhere between 3-4V which I think wouldn’t be as reliable. I’m assuming I’d need a DC/DC buck convertor here. What’s the best way to go about this? Should I use the prototyping deck and integrate the convertor and the Pi there? 
2. Would the 20g of extra thrust from the thrust upgrade bundle work to fly this? The Pi is 9g, the camera is 3g, and the convertor might be another gram or two. I think this adds 
3. If I somehow do all this, realistically, what’s the flight time I’d get from the Crazyflie? This isn’t a high-priority for me (I only need 30-60s flights) but it’d be good to have an idea. 

Thanks!

**Comment by tobbeanton** [comment, marked as answer]
1. In 2018 we did play around a bit with flying a R-Pi-Zero. We made a "adapter" deck as described in [this](https://www.bitcraze.io/2018/08/raspberry-pi-zero-power-deck/) blog post. A prototype deck with a DC/DC converter should work just as well.
2. It will most likely work but if I remember correctly the Pi-Zero is a little bit to big and we made new and longer 3D printed motor mounts where we fitted 8x20 motors. This was enough to lift it and with a 500mAh battery I think it flew for ~3min.
3. You would need a bigger battery, E.g. the tattu 350mAh, but not too big which could add to much weight. Then you could get maybe around 2min hover time, if it mechanically works out...

**Reply by ohharsen** [reply]
Do you still have the stl files for the new motor mounts?

**Reply by ataffanel** [reply]
I do not thing we have any trace of this design left, however you can restart from this openscad project: https://github.com/bitcraze/bitcraze-mechanics/tree/master/motor-mounts/cf2-mount-openscad

**Reply by r-bt** [reply]
@ataffanel Do you still have the electronic design (schematic / layout / gerbers) for the adapter deck?

**Reply by ataffanel** [reply]
We do not have it ready to release. We will investigate what it would take for us to release the kicad project but it might take a couple of weeks.

**Comment by r-bt** [comment]
Hey, @shreeram-murali I'm also considering doing this (integrating a RPi Zero with a Crazyflie with the thrust upgrade kit). I'm curious did you get it to work?

**Reply by shreeram-murali** [reply]
We did! We used the Raspberry Pi 0 to process images from the Raspicam and output measurements that we streamed wirelessly from the Pi. These measurements were then sent to the controlling computer, which would then send control inputs to the Crazyflie. I’m no longer associated with the research group where we did this, but in general, yes, we were able to do this and it worked!

**Reply by r-bt** [reply]
Do you happen to remember how you powered the Raspberry PI?
